### PR TITLE
Add handler for stdClass

### PR DESCRIPTION
**Pull request by Spea**
As mentioned in #51 and #211 I added a new handler to treat stdClass correctly.

### CODE REVIEW

**Comment by Spea** [comment]
ping :)

**Comment by Spea** [comment]
@schmittjoh do you need something else to get this PR merged? would be really nice

**Comment by alexdesignworks** [comment]
@schmittjoh
Can we please have this merged. There are like 10 of issues open regarding this.

**Comment by hshn** [comment]
+1

**Comment by hidenorigoto** [comment]
:+1:

**Comment by qckanemoto** [comment]
+1

**Comment by mathroc** [comment]
:+1:

**Comment by kcivey** [comment]
:+1:

File: tests/JMS/Serializer/Tests/Serializer/BaseSerializationTest.php [hunk 1]
```diff
@@ -29,9 +30,11 @@
 use JMS\Serializer\Tests\Fixtures\Garage;
 use JMS\Serializer\Tests\Fixtures\InlineChildEmpty;
 use JMS\Serializer\Tests\Fixtures\NamedDateTimeArraysObject;
+use JMS\Serializer\Tests\Fixtures\ObjectWithStdClass;
 use JMS\Serializer\Tests\Fixtures\Tree;
 use JMS\Serializer\Tests\Fixtures\VehicleInterfaceGarage;
 use PhpCollection\Sequence;
+use stdClass;
```
Comment (stof) [inline tests/JMS/Serializer/Tests/Serializer/BaseSerializationTest.php]: This bundle does not import classes of the global namespace currently

**Comment by goetas** [comment]
Sorry for the terribly long feedback loop

Closing. This is a feature that can be integrated with a simple class handler (exactly how is done here) using user-land code

**Comment by mathroc** [comment]
@goetas it can indeed be integrated in user-land code but it's odd not to have it in the core bundle. I don't get why `stdClass` does not deserves to be supported directly, it's one of the core structure of PHP, almost on the same level as `array`
it could obviously be made as an external package but it would still be a PITA to require this package and then register the handler. 

Do you mind explaining why you think it should not be supported ?

thank you

**Comment by goetas** [comment]
It is an specific use case and i prefer to have in the core as less things as possible.

The serialization/deserialization of `stdClass` is a special case that some people can have, but IMO is a bad practice.

The model should be typed, express behaviors, expose logic (not implementation).... and `stdClas` is a way to expose inner implementation.
For the same reason many libraries are getting rid of "configuration arrays"   preferring classes. 

So i have two suggestions here: 
- get rid of `stdClass` in your code
- implement this class handler in a user package

**Pull request closed without merging**